pkg/sbproxy/reconcile/reconcile_brokers.go

```diff
@@ -28,37 +28,50 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-// reconcileBrokers attempts to reconcile the current brokers state in the platform (existingBrokers)
-// to match the desired broker state coming from the Service Manager (payloadBrokers).
-func (r *resyncJob) reconcileBrokers(ctx context.Context, existingBrokers, payloadBrokers []*platform.ServiceBroker) {
+// to match the desired broker state coming from the Service Manager (desiredBrokers).
+func (r *resyncJob) reconcileBrokers(ctx context.Context, existingBrokers, desiredBrokers []*platform.ServiceBroker) {
 	brokerKeyMap := indexBrokers(existingBrokers, func(broker *platform.ServiceBroker) (string, bool) {
 		return getBrokerKey(broker), true
 	})
 	proxyBrokerIDMap := indexBrokers(existingBrokers, func(broker *platform.ServiceBroker) (string, bool) {
-		if strings.HasPrefix(broker.BrokerURL, r.proxyPath) {
-			return broker.BrokerURL[strings.LastIndex(broker.BrokerURL, "/")+1:], true
+		if strings.HasPrefix(broker.BrokerURL, r.smPath) {
+			return brokerIDFromURL(broker.BrokerURL), true
+		}
+		return "", false
+	})
+	orphanProxyBrokerMap := indexBrokers(existingBrokers, func(broker *platform.ServiceBroker) (string, bool) {
+		if broker.BrokerURL == fmt.Sprintf(r.proxyPathPattern, brokerIDFromURL(broker.BrokerURL)) {
+			return broker.BrokerURL, true
 		}
 		return "", false
 	})
 
-	for _, payloadBroker := range payloadBrokers {
-		payloadBroker := payloadBroker
-		existingBroker, alreadyProxified := proxyBrokerIDMap[payloadBroker.GUID]
-		delete(proxyBrokerIDMap, payloadBroker.GUID)
-
-		platformBroker, shouldBeProxified := brokerKeyMap[getBrokerKey(payloadBroker)]
+	for _, desiredBroker := range desiredBrokers {
+		desiredBroker := desiredBroker
+		existingBroker, alreadyProxified := proxyBrokerIDMap[desiredBroker.GUID]
+		delete(proxyBrokerIDMap, desiredBroker.GUID)
 
 		if alreadyProxified {
-			if existingBroker.Name != r.brokerProxyName(payloadBroker) { // broker name has been changed in the platform
-				r.updateBrokerRegistration(ctx, existingBroker.GUID, payloadBroker)
+			if existingBroker.Name != r.brokerProxyName(desiredBroker) { // broker name has been changed in the platform
+				r.updateBrokerRegistration(ctx, existingBroker.GUID, desiredBroker)
 				continue
 			}
 			r.fetchBrokerCatalog(ctx, existingBroker)
 		} else {
+			platformBroker, shouldBeProxified := brokerKeyMap[getBrokerKey(desiredBroker)]
+
+			if !shouldBeProxified {
```

Comment:
having 3 check for `shouldBeProxified` doesn't look great
Why not remove `orphanProxyBrokerMap` and add logic in `proxyBrokerIDMap` such as
```golang
....
if broker.BrokerURL == fmt.Sprintf(r.proxyPathPattern, brokerID) {
	return brokerID, true
}
...
``` 

Then on line 55 you can add ` || !strings.HasPrefix(existingBroker.BrokerURL, r.smPath)` where this if can be extracted